packages/wrangler/src/pages/download-config.ts

```diff
@@ -161,32 +183,105 @@ async function toEnvironment(
 	return configObj;
 }
 async function writeWranglerToml(toml: RawEnvironment) {
+	let tomlString = TOML.stringify(toml as TOML.JsonMap);
+
+	// Remove indentation from the start of lines, as this isn't common across TOML examples, and causes user confusion
+	tomlString = tomlString
+		.split("\n")
+		.map((line) => line.trimStart())
+		.join("\n");
+
 	// Pages does not support custom wrangler.toml locations, so always write to ./wrangler.toml
 	await writeFile(
 		"wrangler.toml",
-		[
-			`# Generated by Wrangler on ${new Date()}`,
-			TOML.stringify(toml as TOML.JsonMap),
-		].join("\n")
+		`# Generated by Wrangler on ${new Date()}\n${tomlString}`
 	);
 }
 
+function simplifyEnvironments(
+	preview: RawEnvironment,
+	production: RawEnvironment
+): {
+	topLevel: RawEnvironment;
+	preview?: RawEnvironment;
+	production?: RawEnvironment;
+} {
+	const topLevel = { ...preview };
+	// If the environments are equal, don't include a production override
+	if (JSON.stringify(preview) === JSON.stringify(production)) {
+		// Don't include extraneous placement. No need to check the production setting since it's equal to preview
+		if (topLevel.placement?.mode === "off") {
+			delete topLevel.placement;
+		}
+		return { topLevel };
+	}
+	// Remove duplication for inheritable keys (https://developers.cloudflare.com/pages/functions/wrangler-configuration/#inheritable-keys)
+	if (preview.compatibility_date === production.compatibility_date) {
+		delete production.compatibility_date;
+		delete preview.compatibility_date;
+	}
+	if (
+		JSON.stringify(preview.compatibility_flags) ===
+		JSON.stringify(production.compatibility_flags)
+	) {
+		delete production.compatibility_flags;
+		delete preview.compatibility_date;
+	}
+
+	if (
+		JSON.stringify(preview.placement) === JSON.stringify(production.placement)
+	) {
+		delete production.placement;
+		delete preview.compatibility_date;
+
+		// Don't include extraneous placement
+		if (topLevel.placement?.mode === "off") {
+			delete topLevel.placement;
+		}
+	}
+	if (JSON.stringify(preview.limits) === JSON.stringify(production.limits)) {
+		delete production.limits;
+		delete preview.limits;
+		return { topLevel, production };
+	}
```

Comment:
what about `send_metrics`? Pages supports those as well

Reply:
Is `send_metrics` ever sent to the API? I thought it was just recorded locally in the file

Reply:
hmmm...I have no idea TBH